smithy-model/src/main/java/software/amazon/smithy/model/validation/validators/ResourceOperationInputOutputValidator.java

```diff
@@ -0,0 +1,309 @@
+            validateOperationInputOutput(model, memberPropertyIndex, operationIndex, resource, operation,
+                                         "put", events);
+        });
+
+        resource.getCreate().flatMap(model::getShape).flatMap(Shape::asOperationShape).ifPresent(operation -> {
+            Set<String> identifierMembers = new HashSet<>(identifierBindingIndex
+                    .getAllOperationIdentifiers(resource, operation));
+            propertiesInOperations.addAll(getAllOperationProperties(model, memberPropertyIndex, identifierMembers,
+                    operationIndex, operation));
+            validateOperationInputOutput(model, memberPropertyIndex, operationIndex, resource, operation,
+                                         "create", events);
+        });
+
+        resource.getRead().flatMap(model::getShape).flatMap(Shape::asOperationShape).ifPresent(operation -> {
+            Set<String> identifierMembers = new HashSet<>(identifierBindingIndex
+                    .getAllOperationIdentifiers(resource, operation));
+            propertiesInOperations.addAll(getAllOperationProperties(model, memberPropertyIndex, identifierMembers,
+                    operationIndex, operation));
+            validateOperationInputOutput(model, memberPropertyIndex, operationIndex, resource, operation,
+                                         "read", events);
+        });
+
+        resource.getUpdate().flatMap(model::getShape).flatMap(Shape::asOperationShape).ifPresent(operation -> {
+            Set<String> identifierMembers = new HashSet<>(identifierBindingIndex
+                    .getAllOperationIdentifiers(resource, operation));
+            propertiesInOperations.addAll(getAllOperationProperties(model, memberPropertyIndex, identifierMembers,
+                    operationIndex, operation));
+            validateOperationInputOutput(model, memberPropertyIndex, operationIndex, resource, operation,
+                                         "update", events);
+        });
+
+        resource.getDelete().flatMap(model::getShape).flatMap(Shape::asOperationShape).ifPresent(operation -> {
+            Set<String> identifierMembers = new HashSet<>(identifierBindingIndex
+                    .getAllOperationIdentifiers(resource, operation));
+            propertiesInOperations.addAll(getAllOperationProperties(model, memberPropertyIndex, identifierMembers,
+                    operationIndex, operation));
+            validateOperationInputOutput(model, memberPropertyIndex, operationIndex, resource, operation,
+                                         "delete", events); });
+
+        Set<String> definedProperties = new HashSet<>(resource.getProperties().keySet());
+        definedProperties.removeAll(propertiesInOperations);
+        for (String propertyNotInLifecycleOp : definedProperties) {
+            events.add(error(resource, String.format("Resource shape's `%s` property is not mapped to an input or "
```

Comment:
Can this be rewritten to not use a possessive? Something more like "Resource property `x` ..."